src/Voron/Impl/Paging/CryptoPager.cs

```diff
@@ -263,6 +284,7 @@ public override void BreakLargeAllocationToSeparatePages(IPagerLevelTransactionS
                     buffer.Modified = true;
                 }
 
+                buffer.AddRef();
```

Comment:
You are incrementing the reference even for buffers with `OriginalSize = 0`. But in `ForgetAboutPage()` and `TxOnDispose()` you don't decrement reference when:

```
 if (CanReturnBuffer(buffer) == false)
                return;
```

while `CanReturnBuffer()` checks exactly for that:

https://github.com/ravendb/ravendb/blob/0ccf65d89eef29be3faadf0bf567b6839ec75a41/src/Voron/Impl/Paging/CryptoPager.cs#L372-L379

This means that we'll add references for that pages but never decrement its usage.

Reply:
changed